### ISSUE TITLE
Decision Tree 'max_depth' parameter type 

### ISSUE THREAD

**Issue by teresaconc** [automl/auto-sklearn#569]
For the decision tree classifier the **max_depth** parameter is being set as float:
```
max_depth = UniformFloatHyperparameter('max_depth', 0., 2., default_value=0.5)
```
Shouldn't it be an int? In the oficial sklearn documentation for DecisionTreeClassifier it is stated that it it should be either an Int or None 

> max_depth : int or None, optional (default=None)

**Comment by mfeurer** [comment]
The hyperparameter `max_depth` is indeed confusing as it is a factor. The exact definition is:
```
        # Heuristic to set the tree depth
        if check_none(self.max_depth):
            max_depth = self.max_depth = None
        else:
            num_features = X.shape[1]
            self.max_depth = int(self.max_depth)
            max_depth = max(1, int(np.round(self.max_depth * num_features, 0)))
```

Do you maybe want to contribute to the repository and change the parameter name to make it less confusing?

**Comment by diogo-bastos** [comment]
Hello, may I contribute on this issue? What do you think would be a good name? Something like max_deph_factor? I would need to change `decision_tree.py`, right?

**Comment by teresaconc** [comment]
@mfeurer , the most confusing part is that the code is implementing a sklearn Decision Tree which has a parameter named max_depth that has a different meaning. This is problematic if someone wishes to implement any model from scratch with the given "show_models()" parameters.

I think the perfect solution would probably be to show the end-user the "sklearn" parameter, as in, if the depth_ratio is 0.5 of the feature space size, and the dataset has 30 features, then the parameter shown would be 15. However, I understand that it may go against the core of your idea for this parameter, which is to dynamically provide the best value given the dataset characteristics. In this way it would be a static one.

So the alternative would be to change the name to something like @diogo-bastos suggested: _max_depth_factor_ or _max_depth_ratio_

**Comment by mfeurer** [comment]
> Hello, may I contribute on this issue?

Definitely.

>  Something like max_deph_factor?

Yes, that's a great name

> I would need to change decision_tree.py, right?

Yes, for both regression and classification.

@teresaconc I totally agree that this is confusing right now. And yes, we require the hyperparameters to be independent of the data characteristics to do meta-learning.

**Comment by diogo-bastos** [comment]
I'll start working on this tomorrow then!

**Comment by diogo-bastos** [comment]
Pull request #590 addresses this issue, please review it when you can!

**Comment by mfeurer** [comment]
Will be fixed in the next release via #590 by @diogo-bastos
